### DISCUSSION TITLE
Window size meaning & Setting Distance Threshold

### DISCUSSION THREAD

**Discussion by phant0mon** [TDAmeritrade/stumpy#482] [Q&A] [answered]
Stumpy is awesome! It simplified many difficulties I faced when I implemented time series anamoly detection, but I have multiple questions. 
1. My data consists of 15 mins difference from one data point to the other, how can I set the window size to represent 2 or 3 hours for example? (I really don't understand how window size works in stumpy as I saw some tutorials from here).
2. How can I set a threshold of the distance, so the results show me multiple anomalies, not only the highest distance?

P.S. I am new to the field and I hope you can help me! 
Many thanks!

**Comment by JaKasb** [comment, marked as answer]
1. The windowsize is the length of the sub-sequences. Matrix Profile performs pairwise comparison of all possible sub-sequences.
   windowsize = number of samples
   2 hours / 15 minutes = 2h / 0.25h = 8 samples
2. The problem is not setting a threshold but choosing a specific value for the threshold.
   You can formulate an optimization problem, if you know which performance metric you want to maximize.
   f(data, threshold) -> performance metric
   maximize f(data, threshold) by varying threshold.
   Usually you also need annotated labelled data (ground truth) for the performance metric.

**Reply by phant0mon** [reply]
Thank you so much for the clear answer!
